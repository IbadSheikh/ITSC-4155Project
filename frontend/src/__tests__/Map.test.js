import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import ReviewMap from '../components/Map'; // Ensure the path is correct
import { BrowserRouter as Router } from 'react-router-dom';

// Mock dependencies
jest.mock('react-leaflet', () => ({
  MapContainer: ({ children }) => <div>{children}</div>,
  TileLayer: () => <div>TileLayer</div>,
  Marker: ({ children }) => <div>{children}</div>,
  Popup: ({ children }) => <div>{children}</div>,
  useMap: () => ({ on: jest.fn() }),
}));

jest.mock('../components/CreateReview', () => () => <div>CreateReview Component</div>);

global.navigator.geolocation = {
  getCurrentPosition: jest.fn().mockImplementationOnce((success) =>
    success({ coords: { latitude: 40.7128, longitude: -74.0060 } })
  ),
};

describe('ReviewMap Component', () => {
  test('renders without crashing', () => {
    render(
      <Router>
        <ReviewMap />
      </Router>
    );
    expect(screen.getByText(/TileLayer/i)).toBeInTheDocument();
  });

  test('renders error message when fetch fails', async () => {
    // Mock the fetchReviews function to simulate a failed fetch
    jest.spyOn(global, 'fetch').mockImplementationOnce(() =>
      Promise.reject(new Error('Network error'))
    );

    render(
      <Router>
        <ReviewMap />
      </Router>
    );

    await waitFor(() => expect(screen.getByText(/Error fetching reviews/i)).toBeInTheDocument());
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <Router>
        <ReviewMap />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
