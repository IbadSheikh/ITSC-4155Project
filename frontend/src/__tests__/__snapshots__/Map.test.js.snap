// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewMap Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        TileLayer
      </div>
    </div>
  </div>
</DocumentFragment>
`;
